voxblox/src/utils/voxel_utils.cc

```diff
@@ -18,6 +28,12 @@ void mergeVoxelAIntoVoxelB(const TsdfVoxel& voxel_A, TsdfVoxel* voxel_B) {
                                            voxel_B->color, voxel_B->weight);
 
     voxel_B->weight = combined_weight;
+  } else {
+    // If the combined voxel weight is smaller or equal to 0, and one of voxel
```

Comment:
In the comment you mention that "...if the combined voxel weight is smaller or equal to 0 **and** one of he voxel weights is smaller than 0..." but you do not check for the latter statement.

I guess that if the combined voxel weight is below zero then anyway one of the voxel weights must be negative. Maybe adjust this in the comment (this and the suggestion below).

```suggestion
    // If the combined voxel weight is smaller or equal to 0, this means that one of voxel
```

Reply:
good point, thanks for the suggestion! yes, that's what I wanted to say, I'll change the formulation